bson/__init__.py

```
@@ -1308,7 +1308,7 @@ def encode(
         """
         return cls(encode(document, check_keys, codec_options))
 
-    def decode(self, codec_options: CodecOptions = DEFAULT_CODEC_OPTIONS) -> _DocumentType:  # type: ignore[override]
+    def decode(self, codec_options: "CodecOptions[_DocumentType]" = DEFAULT_CODEC_OPTIONS) -> _DocumentType:  # type: ignore[override,assignment]
```

The "assignment" ignore is required to silence this error: 
```
bson/__init__.py: note: In member "decode" of class "BSON":
bson/__init__.py:1311: error: Incompatible default for argument "codec_options" (default has type "CodecOptions[MutableMapping[str, Any]]", argument
has type "CodecOptions[_DocumentType]")  [assignment]
        def decode(self, codec_options: "CodecOptions[_DocumentType]" = DEFAULT_CODEC_OPTIONS) -> _DocumentType:  # type: ignore[override]
```